OSDOCS#10177: Updates to Terraform files (#77311)
openshift/openshift-docs: rosa_install_access_delete_clusters/terraform/rosa-classic-creating-a-cluster-with-customizations-terraform.adoc

:_mod-docs-content-type: ASSEMBLY
[id="rosa-classic-creating-a-cluster-with-customizations-terraform-classic"]
= Customizing a ROSA cluster with Terraform
include::_attributes/attributes-openshift-dedicated.adoc[]
:context: rosa-classic-creating-a-cluster-with-customizations-terraform-classic

toc::[]

You can use Terraform to create a customized {product-title} (ROSA) cluster by using Terraform to customize the cluster's details.

[NOTE]
====

> This guide assumes a basic understanding of ROSA. See the additional resources for conceptual information about ROSA clusters.

Reviewer: ```suggestion
This guide assumes a basic understanding of ROSA. See "Additional resources" for conceptual information about ROSA clusters.
```

====

.Prerequisites

* You have generated and stored an link:https://console.redhat.com/openshift/token[offline {cluster-manager-first} token].
* You have an AWS account with the correct permissions.
* You completed the xref:../../rosa_planning/rosa-cloud-expert-prereq-checklist.adoc[Prerequisites checklist for deploying ROSA using STS]
* You completed the setup steps in xref:../../rosa_planning/rosa-understanding-terraform.adoc[Preparing Terraform to install ROSA clusters]

> * You have installed link:https://developer.hashicorp.com/terraform/install[the `terraform` CLI tool].

Reviewer: ```suggestion
* You have installed the link:https://developer.hashicorp.com/terraform/install[`terraform` CLI tool].
```

[id="rosa-classic-creating-a-cluster-with-customizations-terraform-creating"]
== Creating a ROSA Classic cluster using Terraform

> The cluster creation process outlined below shows how to use Terraform to create your account-wide IAM roles and a default ROSA cluster with a managed OIDC configuration.

Reviewer: ```suggestion
The following cluster creation process shows how to use Terraform to create your account-wide IAM roles and a default ROSA cluster with a managed OIDC configuration.
```
